web/src/lib/stores/asset-interaction.store.ts

```diff
@@ -62,18 +63,28 @@ function createAssetInteractionStore() {
 		isViewingAssetStoreState.set(isViewing);
 	};
 
+	const dateFormat: Intl.DateTimeFormatOptions = {
+		weekday: 'short',
+		month: 'short',
+		day: 'numeric',
+		year: 'numeric'
+	};
+
 	const navigateAsset = async (direction: 'next' | 'previous') => {
 		// Flatten and sort the asset by date if there are new assets
 		if (assetSortedByDate.length === 0 || savedAssetLength !== _assetGridState.assets.length) {
-			assetSortedByDate = sortBy(_assetGridState.assets, (a) => a.fileCreatedAt);
+			assetSortedByDate = sortBy(
+				_assetGridState.assets,
+				(a) => new Date(a.fileCreatedAt).toLocaleDateString(locale, dateFormat)[0]
```

Comment:
Hmm, I think we should be sorting it by the underlying date, not by the display date string. What if a local uses `dd/mm/yy` then December 1st would show up before January 2nd.

Reply:
Thank you for your input!
I totally agree and have also noticed that I misinterpreted the code in the asset date group as that sorts the groups and not the images inside of a group. I've reverted my changes and instead added sorting by creation date in the grid view.